client/extensions/woocommerce/state/sites/settings/general/reducer.js

```diff
@@ -4,13 +4,30 @@
 import { createReducer } from 'state/utils';
 import { LOADING } from 'woocommerce/state/constants';
 import {
+	WOOCOMMERCE_CURRENCY_UPDATE,
+	WOOCOMMERCE_CURRENCY_UPDATE_SUCCESS,
 	WOOCOMMERCE_SETTINGS_GENERAL_REQUEST,
 	WOOCOMMERCE_SETTINGS_GENERAL_REQUEST_SUCCESS,
 } from 'woocommerce/state/action-types';
 
 // TODO: Handle error
 
 export default createReducer( {}, {
+	[ WOOCOMMERCE_CURRENCY_UPDATE ]: ( state ) => {
+		// TODO: Return some sort of saving indicator
+		return state;
+	},
+
+	[ WOOCOMMERCE_CURRENCY_UPDATE_SUCCESS ]: ( state, { data } ) => {
+		const settings = state || [];
+		const newSettings = settings.map( ( setting ) => {
```

Comment:
If settings is empty from above, or the currency setting doesn't exist in the array, our new value wouldn't get set.